runtime/bigarray.js

```diff
@@ -498,6 +498,35 @@ function caml_ba_get_generic(ba, index) {
     return ba.get(caml_js_from_array(index));
 }
 
+//Provides: caml_ba_uint8_get16
+function caml_ba_uint8_get16(ba, i0) {
+    var b1 = ba.get1(i0);
+    var b2 = ba.get1(i0+1) << 8;
+    return (b1 | b2);
+}
+
+//Provides: caml_ba_uint8_get32
+function caml_ba_uint8_get32(ba, i0) {
+    var b1 = ba.get1(i0);
+    var b2 = ba.get1(i0+1) << 8;
+    var b3 = ba.get1(i0+2) << 16;
+    var b4 = ba.get1(i0+3) << 24;
+    return (b1 | b2 | b3 | b4);
+}
+
+//Provides: caml_ba_uint8_get64
+function caml_ba_uint8_get64(ba, i0) {
+    var b1 = ba.get1(i0);
+    var b2 = ba.get1(i0+1) << 8;
+    var b3 = ba.get1(i0+2) << 16;
+    var b4 = ba.get1(i0+3) << 24;
+    var b5 = ba.get1(i0+4) << 32;
+    var b6 = ba.get1(i0+5) << 40;
+    var b7 = ba.get1(i0+6) << 48;
+    var b8 = ba.get1(i0+7) << 56;
+    return (b1 | b2 | b3 | b4 | b5 | b6 | b7 | b8);
```

Comment:
integers are 32bit in javascript. You should return an int64 here.
see https://github.com/ocsigen/js_of_ocaml/blob/master/runtime/int64.js